### PR TITLE
[#12048] Migrate CreateFeedbackQuestionAction

### PR DESCRIPTION
**Pull request by hhdqirui**
Part of #12048 

- Remove the field `questionText` in `FeedbackQuestion` because initially question details are stored as a json string in `questionText`, and now we have all the different types of question inheriting `FeedbackQuestion` and the question details are stored in each subclass. The `questionText` in the parent class is redundant.
- `JsonConverter` for `questionDetails` is changed to one not using generic as the one with generic is not working currently, and it will be updated in the future.
- Data bundle for integration tests is updated to include some but not all feedback questions from the original data bundle. The rest is to be updated in the future.
- One architecture test, `testArchitecture_uiShouldNotTouchStorage` is updated.
- Integration test for this action is to be added after `GetFeedbackQuestionAction` is migrated.

### CODE REVIEW

File: src/it/resources/data/DataBundleLogicIT.json [hunk 1]
```diff
@@ -192,7 +192,33 @@
...
+      "feedbackSession": {
+        "id": "00000000-0000-4000-8000-000000000701"
+      },
+      "questionDetails": {
+        "questionType": "TEXT",
+        "questionText": "What is the best selling point of your product?"
+      },
+      "description": "This is a text question.",
+      "questionNumber": 1,
+      "questionType": "TEXT",
```
Comment (samuelfangjw) [inline src/it/resources/data/DataBundleLogicIT.json:207]: questionType seems to be repeated here. possible to remove this redundancy?

File: src/main/java/teammates/ui/webapi/CreateFeedbackQuestionAction.java [hunk 2]
```diff
@@ -26,9 +36,17 @@ void checkSpecificAccessControl() throws UnauthorizedAccessException {
         String courseId = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ID);
         String feedbackSessionName = getNonNullRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_NAME);
         InstructorAttributes instructorDetailForCourse = logic.getInstructorForGoogleId(courseId, userInfo.getId());
+        if (instructorDetailForCourse != null) {
+            gateKeeper.verifyAccessible(instructorDetailForCourse,
+                    getNonNullFeedbackSession(feedbackSessionName, courseId),
+                    Const.InstructorPermissions.CAN_MODIFY_SESSION);
+            return;
+        }
```
Comment (samuelfangjw) [inline src/main/java/teammates/ui/webapi/CreateFeedbackQuestionAction.java:44]: I believe we need to use the isCourseMigrated method here.

There's a case that's not considered here. After migration, datastore entity is still non-null. However, source of truth is now the sql entity, not the datastore one.

File: src/main/java/teammates/ui/webapi/CreateFeedbackQuestionAction.java [hunk 3]
```diff
@@ -38,39 +56,166 @@ public JsonResult execute() throws InvalidHttpRequestBodyException {
...
+                    request.getQuestionNumber(),
+                    request.getQuestionDescription(),
+                    request.getQuestionDetails().getQuestionType(),
+                    request.getGiverType(),
+                    request.getRecipientType(),
+                    request.getNumberOfEntitiesToGiveFeedbackTo(),
+                    request.getShowResponsesTo(),
+                    request.getShowGiverNameTo(),
+                    request.getShowRecipientNameTo(),
+                    request.getQuestionDetails()
```
Comment (samuelfangjw) [inline src/main/java/teammates/ui/webapi/CreateFeedbackQuestionAction.java:74]: Maybe it's time we consider a factory method for questions and responses. Can have a static method (maybe makeQuestion or similar?) in FeedbackQuestions that takes in the params and the question type and returns the subclass.

File: src/main/java/teammates/storage/sqlentity/responses/FeedbackConstantSumResponse.java [hunk 4]
```diff
@@ -41,6 +41,6 @@ public String toString() {
      */
     @Converter
     public static class FeedbackConstantSumResponseDetailsConverter
-            extends JsonConverter<FeedbackConstantSumResponseDetails> {
+            extends FeedbackQuestionDetailsConverter {
```
Comment (cedricongjh) [inline src/main/java/teammates/storage/sqlentity/responses/FeedbackConstantSumResponse.java:44]: if i'm not wrong, these are response details, and not question details and hence the converter here is incorrect (need to create a `FeedbackResponseDetailsConverter`)

File: src/it/resources/data/DataBundleLogicIT.json (hunk 1 above)
Comment (hhdqirui) [reply src/it/resources/data/DataBundleLogicIT.json]: Yes! Thanks for the catch!

File: src/main/java/teammates/ui/webapi/CreateFeedbackQuestionAction.java (hunk 3 above)
Comment (hhdqirui) [reply src/main/java/teammates/ui/webapi/CreateFeedbackQuestionAction.java]: Sure! Thanks for the advice!

File: src/main/java/teammates/storage/sqlentity/responses/FeedbackConstantSumResponse.java (hunk 4 above)
Comment (hhdqirui) [reply src/main/java/teammates/storage/sqlentity/responses/FeedbackConstantSumResponse.java]: Yes! Thanks for the catch!

**Review by samuelfangjw — APPROVED** [review]

**Pull request merged**